rotkehlchen/globaldb/handler.py

```diff
@@ -263,11 +263,17 @@ def retrieve_assets(filter_query: 'AssetsFilterQuery') -> Tuple[List[Dict[str, A
                         'started': entry[6],
                     })
                 elif asset_type == AssetType.EVM_TOKEN:
+                    # make additional query to get underlying tokens
+                    underlying_tokens = GlobalDBHandler().fetch_underlying_tokens(
+                        cursor=cursor,
+                        parent_token_identifier=entry[0],
+                    )
```

Comment:
This generally is bad because we are doing one extra call to the database per evm token in the query result. 

I believe it would be better if you make only one extra call to the database with something like "select the underlying token properties of the evm tokens in (fetch the identifiers of only only the evm tokens with the query filter)" then deserialize in a hash map by the evm_parten token identifier `Dict[parent_identifier, underlying token info]` and then use it here.

As it is now you are doing 1 query per evm token and the other way you make only 1